src/Admin/DevToolsUserAccess.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Class DevToolsUserAccess.
+ *
+ * @since 1.6.0
+ *
+ * @package AMP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_User;
+
+/**
+ * Class DevToolsUserAccess
+ *
+ * @since 1.6.0
+ */
+final class DevToolsUserAccess implements Delayed, Service, Registerable {
+
+	/**
+	 * User meta key enabling or disabling developer tools.
+	 *
+	 * @var string
+	 */
+	const USER_FIELD_DEVELOPER_TOOLS_ENABLED = 'amp_dev_tools_enabled';
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Runs on instantiation.
+	 *
+	 * @action rest_api_init
+	 */
+	public function register() {
+		register_rest_field(
+			'user',
+			self::USER_FIELD_DEVELOPER_TOOLS_ENABLED,
+			[
+				'get_callback'    => [ $this, 'rest_get_dev_tools_enabled' ],
+				'update_callback' => [ $this, 'rest_update_dev_tools_enabled' ],
+				'schema'          => [
+					'description' => __( 'Whether the user has enabled dev tools.', 'amp' ),
+					'type'        => 'boolean',
+				],
+			]
+		);
+	}
+
+	/**
+	 * Provides the user's dev tools enabled setting.
+	 *
+	 * @param array $user Array of user data prepared for REST.
+	 * @return null|boolean Whether tools are enabled for the user, or null if the option has not been set.
+	 */
+	public function rest_get_dev_tools_enabled( $user ) {
+		$meta = get_user_meta( $user['id'] );
+
+		if ( is_array( $meta ) && array_key_exists( self::USER_FIELD_DEVELOPER_TOOLS_ENABLED, $meta ) ) {
+			return boolval(
+				is_array( $meta[ self::USER_FIELD_DEVELOPER_TOOLS_ENABLED ] ) && ! empty( $meta[ self::USER_FIELD_DEVELOPER_TOOLS_ENABLED ] )
+					? reset( $meta[ self::USER_FIELD_DEVELOPER_TOOLS_ENABLED ] )
+					: $meta[ self::USER_FIELD_DEVELOPER_TOOLS_ENABLED ]
+			);
+		}
```

Comment:
Per below, this could then be simplified to:

```suggestion
		$meta_value = get_user_meta( $user['id'], self::USER_FIELD_DEVELOPER_TOOLS_ENABLED, true );
		if ( '' !== $meta_value ) ) {
			return rest_sanitize_boolean( $meta_value );
		}
```

Reply:
Done in 3139da7